src/TraceEvent/Computers/CLRRuntimeActivityComputer.cs

```diff
@@ -0,0 +1,343 @@
+                foreach (var startStopIn in inputStream)
+                {
+                    var startStop = startStopIn;
+                    if (currentPerThreadProcessingState.Count > 0)
+                    {
+                        while ((currentPerThreadProcessingState.Count > 0) && (currentPerThreadProcessingState.Peek().End.CompareTo(startStop.Start) < 0))
+                        {
+                            // Current stack top event finished before this event happened.
+                            var poppedEvent = currentPerThreadProcessingState.Pop();
+                            EventUID lastEventProcessedEnd = poppedEvent.End;
+                            if (currentPerThreadProcessingState.Count > 0)
+                            {
+                                var tempPoppedEvent = currentPerThreadProcessingState.Pop();
+                                tempPoppedEvent.Start = lastEventProcessedEnd;
+                                yield return tempPoppedEvent;
+                                currentPerThreadProcessingState.Push(tempPoppedEvent);
+                            }
+                        }
+                    }
+
+                    startStop.StackDepth = currentPerThreadProcessingState.Count;
+                    yield return startStop;
+                    currentPerThreadProcessingState.Push(startStop);
+                }
+            }
+        }
+
+        RuntimeLoaderStats _startStopData = new RuntimeLoaderStats();
+        public RuntimeLoaderStats StartStopData => _startStopData;
+
+        Dictionary<IdOfIncompleteAction, IncompleteActionDesc> _incompleteJitEvents = new Dictionary<IdOfIncompleteAction, IncompleteActionDesc>();
+        Dictionary<IdOfIncompleteAction, IncompleteActionDesc> _incompleteR2REvents = new Dictionary<IdOfIncompleteAction, IncompleteActionDesc>();
+        Dictionary<IdOfIncompleteAction, IncompleteActionDesc> _incompleteTypeLoadEvents = new Dictionary<IdOfIncompleteAction, IncompleteActionDesc>();
+
+        public Dictionary<int, List<StartStopThreadEventData>> StartStopEvents { get; } = new Dictionary<int, List<StartStopThreadEventData>>();
+
+        public CLRRuntimeActivityComputer(TraceLogEventSource source)
+        {
+            _startStopData.EventSource = source;
+            source.Clr.MethodJittingStarted += Clr_MethodJittingStarted;
+            source.Clr.MethodR2RGetEntryPoint += Clr_MethodR2RGetEntryPoint;
+            source.Clr.MethodLoadVerbose += Clr_MethodLoadVerbose;
+            source.Clr.MethodLoad += Clr_MethodLoad;
+            source.Clr.LoaderAssemblyLoad += Clr_LoaderAssemblyLoad;
+            source.Clr.MethodR2RGetEntryPointStart += Clr_R2RGetEntryPointStart;
+            source.Clr.LoaderTypeLoadStart += Clr_LoaderTypeLoadStart;
+            source.Clr.LoaderTypeLoadStop += Clr_LoaderTypeLoadStop;
+            source.Process();
+            source.Clr.MethodJittingStarted -= Clr_MethodJittingStarted;
```

Comment:
You should not need to unsubscribe these delegates.  The pattern used is to provide a fresh EventSource so that you are sure that other machinery isn't subscribed if you are concerned about this.  I don't think you'll find many, if any, places in where we attempt to unsubscribe in order to re-use the EventSource.